intel-sgx/attestation-tenant/src/main.rs

```
@@ -0,0 +1,115 @@
+    // The tenant requests attestation from the platform's attestation daemon.
+    // The actual signal is arbitrary.
+    let daemon_conn = TcpStream::connect(DAEMON_CONN)?;
+    let mut daemon_buf = BufStream::new(daemon_conn);
+    daemon_buf.write(&b"Request attestation"[..])?;
+
+    // The tenant receives a Quote from the platform's attestation
+    // daemon. This Quote verifies the enclave's self-measurement from its Report.
+    let mut quote = Vec::new();
+    daemon_buf.read_to_end(&mut quote)?;
+
+    // The signed material for the Quoting Enclave's Attestation Key (Quote Header ||
+    // ISV Enclave Report) is retrieved.
+    let att_key_signed_material = dcap_ql::quote::Quote::raw_header_and_body(&quote)?;
+
+    // The hashed material (containing the Attestation Key) signed by the PCK is retrieved.
+    let hashed_reportdata = dcap_ql::quote::Quote::raw_pck_hash(&quote)?;
+
+    // This parses the Quote's signature section.
+    let quote = dcap_ql::quote::Quote::parse(&quote)?;
+    let q_sig = quote.signature::<Quote3SignatureEcdsaP256>()?;
+    let q_enclave_report_sig = q_sig.signature();
+    let q_qe_report = q_sig.qe3_report();
+    let q_qe_report_sig = q_sig.qe3_signature();
+    let q_att_key_pub = q_sig.attestation_public_key();
+    let q_auth_data = q_sig.authentication_data();
+
+    // The Quote's Certification Data contains the PCK Cert Chain and PCK Certificate;
+    // the embedded PCK signs the Attestation Key.
+    let cert_data = q_sig.certification_data::<Qe3CertDataPckCertChain>()?;
+    let quote_pck_leaf_cert = cert_data.leaf_cert;
+
+    // The tenant's PCK certificate chain must be loaded to verify the Quote's PCK Leaf
+    // Certificate. The root certificate in this chain is trusted, since it is provided by the
+    // tenant.
+    let cert_chain_file = env::args()
+        .nth(1)
+        .expect("You must supply the path of a valid PCK certificate chain as the first argument.");
```

I think this check should be moved to the beginning of the program to be able to fail early.

---

Done.